File: Algorithms/Array/python/selectionSort.py
def findSmallest(array):
    minIdx = 0
    for i in range(1, len(array)):
        if array[i] < array[minIdx]:
            minIdx = i
    return minIdx


def selectionSort(arr):
    newArr = []
    for i in range(len(arr)):
        min = findSmallest(arr)
        newArr.append(arr.pop(min))
    return newArr


'''if __name__ == "__main__":'''
print(selectionSort([5, 3, 6, 2, 10]))

File: Algorithms/Array/selectionSort.js
function selectionSort(array) {
  if (array.length < 1) {
    return array;
  }
  for (let i = 0; i < array.length; i++) {
    let min = array[i];
    let minIdx = i;
    for (let j = i + 1; j < array.length; j++) {
      if (array[j] < min) {
        minIdx = j;
        min = array[j];
      }
    }
    // swap indices
    let temp = array[i];
    array[i] = array[minIdx];
    array[minIdx] = temp;
  }

  return array;
}

let array = [8, 5, 2, 9, 5, 6, 3];
console.log(selectionSort(array));
